### PR TITLE
Observability jmx

### PR DESCRIPTION
**Pull request by bonclay7**
### What does this PR do?

This PR provides patterns for Observability on EKS with provided metrics collection with OpenTelemetry and default dashboards on Grafana

Rework of https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/347

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [x] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/observability/eks-java-jmx/data.tf
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/data.tf:11]: these can go into `main.tf` file. We would like to keep one file for examples to keep ti simple

File: examples/observability/eks-java-jmx/dev.tfvars
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/dev.tfvars:2]: Add this to deployment README on what customers needs to do for this variable

File: examples/observability/eks-java-jmx/providers.tf
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/providers.tf:23]: move this to `main.tf`

File: examples/observability/eks-java-jmx/variables.tf
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/variables.tf:2]: Add `type` and `description`

File: examples/observability/eks-java-jmx/variables.tf
Comment (vara-bonthu) [inline examples/observability/eks-java-jmx/variables.tf:8]: You can just these two variables as expected and remove the `tfvars` file to further simplify

File: modules/kubernetes-addons/aws-observability-pattern-jmx/dashboards/default.json
Comment (vara-bonthu) [inline modules/kubernetes-addons/aws-observability-pattern-jmx/dashboards/default.json:21]: Should the Dashboard be in examples?

**Review by vara-bonthu — COMMENTED** [review]
Thanks @bonclay7! Minor comments

We just to run E2E on this before the merge

File: examples/observability/eks-java-jmx/dev.tfvars
Comment (saaish) [reply examples/observability/eks-java-jmx/dev.tfvars:2]: added

File: examples/observability/eks-java-jmx/providers.tf
Comment (saaish) [reply examples/observability/eks-java-jmx/providers.tf:23]: Moved

File: modules/kubernetes-addons/adot-collector-java/locals.tf
Comment (bryantbiggs) [inline modules/kubernetes-addons/adot-collector-java/locals.tf:8]: I didn't do a full comparison but just curious why the local/custom helm chart and not the one provided by opentelemetry? https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts

**Review by bryantbiggs — APPROVED** [review]
Looks good enough to ship

**Review by Zvikan — APPROVED** [review]
LGTM assuming e2e/ local tests ran successfully! ty

**Pull request merged**
